app/src/main/java/com/pmatuki/wowpapers/view/detail/DetailScreen.kt

```diff
@@ -55,47 +62,26 @@ internal fun DetailScreen(
         downloadWallpaper(wallpaperUrl)
     }
 
-    when (state) {
-        DetailViewState.Loading -> {
-            ProgressBar()
-        }
-        DetailViewState.Empty -> {
-            // Show nothing
-        }
-        is DetailViewState.ErrorLoading -> {
-            DetailScreenBody(wallpaperImage = null, applyEnabled = false)
-            ErrorToast(
-                errorMessageResId = R.string.image_load_fail,
-                snackBarState = scaffoldState.snackbarHostState
-            )
-        }
-        is DetailViewState.Loaded -> {
-            val loadedState = (state as DetailViewState.Loaded)
-            wallpaperDrawable = loadedState.drawableHolder.item as Drawable
-            DetailScreenBody(wallpaperImage = wallpaperDrawable, applyEnabled = true, onApplyClicked = applyWallpaper)
-        }
-        DetailViewState.Applying -> {
-            ErrorToast(
-                errorMessageResId = R.string.applying_wallpaper_wait,
-                snackBarState = scaffoldState.snackbarHostState
-            )
-            DetailScreenBody(wallpaperImage = wallpaperDrawable, applyEnabled = false)
-        }
-        DetailViewState.Applied -> {
-            DetailScreenBody(wallpaperImage = wallpaperDrawable, applyEnabled = true, onApplyClicked = applyWallpaper)
-            ErrorToast(
-                errorMessageResId = R.string.applying_wallpaper_success,
-                snackBarState = scaffoldState.snackbarHostState
-            )
-        }
-        is DetailViewState.ErrorApplying -> {
-            DetailScreenBody(wallpaperImage = wallpaperDrawable, applyEnabled = true, onApplyClicked = applyWallpaper)
-            ErrorToast(
-                errorMessageResId = R.string.applying_wallpaper_error,
-                snackBarState = scaffoldState.snackbarHostState
-            )
+    LaunchedEffect(true) {
```

Comment:
I couldn't find a way to collect events from the Composable without using a side-effect, even if I totally agree `LaunchedEffect(true)` is not appropriate.

@BrunoTiba found in the Orbit MVI source they are using `LaunchedEffect` with the flow and lifecycle owner as keys so I'm thinking about doing the same.

Reply:
After taking a look in some documentations, yes, even Google uses `LaunchedEffect(true)` in some parts of the code, like [here](https://developer.android.com/jetpack/compose/side-effects#rememberupdatedstate).

However, my understanding about Side-Effects (which may be wrong) is that we used it when we need to launch something for the first time during a composition and only re-run it if the key value changes. In the Google's example, shows something after an initial delay. So, in the case of "collecting states", IMHO, this is not a "Side-effect" per se.

I think that states are an important part of the Compose/Declarative UI environment, then it must be treated accordingly. On Alkaa, [I decided to use `collectAsState`](
https://github.com/igorescodro/alkaa/blob/354817f387119560d20b674b30150f3ac36ddaeb/features/task/src/main/java/com/escodro/task/presentation/list/TaskList.kt#L75) which seems like a prettier way to handle. But keep in mind that my `Flow` is cold. In case of hot flows, probably it is best to use `repeatOnLifecycle`.

Please, let me know what both of you think about this.

Reply:
I think the difference in this point is the flow collected here is an event rather than a state (which is collected on line 45 with `val state by stateFlow.collectAsState()`). Receiving this event will show a "one time" effect which in this case is showing a toast, which shouldn't impact the Screen state permanently.